### PR TITLE
Added bpfcc/bcc test

### PR DESCRIPTION
**Pull request by harish-24**
Runs bcc self test. Added support only for Ubuntu.

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

File: generic/bcc.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                            'libedit-dev', 'libllvm3.8', 'llvm-3.8-dev',
+                            'libclang-3.8-dev', 'python', 'zlib1g-dev',
+                            'libelf-dev', 'clang-format-3.8', 'python-netaddr',
+                            'python-pyroute2', 'arping', 'iperf', 'netperf',
+                            'ethtool']:
+                if not smm.check_installed(package) and not smm.install(package):
+                    self.cancel("Failed to install %s, which is needed for"
+                                "the test to be run" % package)
+        else:
+            self.cancel("Test is yet to be supported for other distro")
```
Comment (PraveenPenguin) [inline generic/bcc.py]: i think this decision will take on  starting of test as line of 43
can be like 
```
        if not distro.detect().name.lower() == 'ubuntu':
               self.cancel('Upsupported OS %s' % distro.detect().name.lower())
```

**Comment by harish-24** [comment]
[bcc_test.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1276547/bcc_test.txt)

File: generic/bcc.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        os.makedirs('%s/build' % self.sourcedir)
+        self.builddir = '%s/build' % self.sourcedir
+        os.chdir(self.builddir)
+        process.run('cmake .. -DCMAKE_INSTALL_PREFIX=/usr', shell=True)
+
+        build.make(self.builddir)
+
+    def test(self):
+
+        if build.make(self.builddir, extra_args='test', ignore_status=True):
```
Comment (PraveenPenguin) [inline generic/bcc.py]: are we need to parse debug file to make more pointed about failure

File: generic/bcc.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+from avocado import Test
+from avocado import main
+from avocado.utils import archive, build, process, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Bcc(Test):
+
+    """
+    Self test case of bcc
```
Comment (PraveenPenguin) [inline generic/bcc.py]: can you please add some more info as what this test does

Comment (harish-24) [reply generic/bcc.py]: Sure

File: generic/bcc.py (hunk 1 above)
Comment (harish-24) [reply generic/bcc.py]: sure, will make the change.

File: generic/bcc.py (hunk 2 above)
Comment (harish-24) [reply generic/bcc.py]: I think it is not required as it is selftest will point out the failures.

**Comment by harish-24** [comment]
@PraveenPenguin Made the changes. Thanks for the review.

**Pull request merged**